packages/mobx/src/types/observableannotation.ts

```diff
@@ -38,6 +46,62 @@ function extend_(
     )
 }
 
+function decorate_20223_(
+    this: Annotation,
+    desc,
+    context: ClassAccessorDecoratorContext | ClassFieldDecoratorContext
+) {
+    assert20223DecoratorType(context, ["accessor", "field"])
+
+    const ann = this
+    const { kind, name: key, addInitializer } = context
+
+    // Backwards/Legacy behavior
+    if (kind == "field") {
+        addInitializer(function () {
+            const adm: ObservableObjectAdministration = asObservableObject(this)[$mobx]
+            adm.defineObservableProperty_(
+                key,
+                (this as any)[key],
+                ann.options_?.enhancer ?? deepEnhancer,
+                false
+            )
+        })
+        return
+    }
+
+    // For the most backwards compatibility, this will ensure that the observable is an enumerable "own" prop.
```

Comment:
Opted to go the non-enumerable route. Easy change to go the other way, but figured it would be better to follow ES's approach to `accessor` enumerability. Created a note in the docs.